dirty_cat/datetime_encoder.py

```diff
@@ -166,5 +168,12 @@ def get_feature_names(self, input_features=None) -> List[str]:
         Ensures compatibility with sklearn < 1.0, and returns the output of
         get_feature_names_out.
         """
+        if Version(sklearn_version) >= '1.0':
+            warn(
+                "Following the changes in scikit-learn 1.0, "
+                "get_feature_names is deprecated. "
+                "Use get_feature_names_out instead.",
+                DeprecationWarning
```

Comment:
Use the "stacklevel" argument of "warnings.warn", when you call "warn", where the value has to be chosen for the warning to appear in user code rather than in library code. Typically 1.